File: src/main/java/bigleq/BigLeq.java
package bigleq;

import cspfj.MGACIter;
import cspfj.Solver;
import cspfj.constraint.Constraint;
import cspfj.constraint.semantic.AllDifferent;
import cspfj.constraint.semantic.Gt;
import cspfj.filter.AC3Constraint;
import cspfj.priorityqueues.FibonacciHeap;
import cspfj.priorityqueues.Key;
import cspfj.problem.BitVectorDomain;
import cspfj.problem.Problem;
import cspfj.problem.Variable;

public class BigLeq {
	private static final int NB_VALS = 1000;
	private static final int NB_VARS = 1000;

	public static Problem bigleq(final int nbVars, final int nbVals) {
		final Problem problem = new Problem();

		final int[] vals = new int[nbVals];
		for (int i = nbVals; --i >= 0;) {
			vals[i] = i;
		}
		final Variable[] vars = new Variable[nbVars];
		for (int i = nbVars; --i >= 0;) {
			vars[i] = problem.addVariable("X" + i, new BitVectorDomain(vals));
		}

		for (int i = nbVars - 1; --i >= 0;) {
			problem.addConstraint(new Gt(vars[i + 1], vars[i], false));
		}

		problem.addConstraint(new AllDifferent(vars));

		return problem;
	}

	public static void main(String[] args) {

		final Problem problem = bigleq(NB_VARS, NB_VALS);
		problem.getVariable("X0").remove(0);

		{
			final Solver s = new MGACIter(problem);
			long time = -System.currentTimeMillis();
			s.nextSolution();
			time += System.currentTimeMillis();

			System.out.println(time);
		}

	}
}

File: src/main/scala/bigleq/BigLeq.scala
package bigleq

import concrete.constraint.semantic.Gt
import concrete.Problem
import cspom.CSPOM
import CSPOM._
import concrete.CSPOMDriver._
import cspom.variable.IntVariable

object BigLeq extends App {
  val NB_VALS = 2000;
  val NB_VARS = 2000;

  def bigleq(nbVars: Int, nbVals: Int) = CSPOM {
    val vars = (1 to nbVars).map {
      case i if i == 2 => IntVariable(2 to nbVals) as ("V" + i)
      case i => IntVariable(1 to nbVals) as ("V" + i)
    }

    for (v <- vars.sliding(2)) {
      ctr(v(1) > v(0))
    }

    ctr(allDifferent(vars: _*))

  }

  //Solver.loggerLevel = "FINE"
  //val problem = bigleq(NB_VARS, NB_VALS);
  //problem.variable("X0").dom.remove(0);

  //ParameterManager("dummy.filter") = classOf[AC3]

  //for (i <- List(50, 100, 200, 500, 1000, 2000, 5000)) {

  val i = 400
  val problem = bigleq(i, i)
  //xml.XML.save("bigleq-" + i + ".xml", XCSPWriter(problem))

  //}

  //val concrete = ProblemGenerator.generate(problem)

  //println(concrete)
  //val s = new DummySolver(concrete);

  // val (result, time) = StatisticsManager.time(s.nextSolution)
  //println(concrete)

  //println(result)
  //println(time);

}
